### ISSUE TITLE
Indel final step error Error in plot.window(...) : need finite 'ylim' values

### ISSUE THREAD

**Issue by danchubb** [skandlab/MutSpot#9]
Hi,

I've ran an SNV and indel run of mutspot. The SNV results completed ok but it crashed near the end of the indel results. 

I get 

![image](https://user-images.githubusercontent.com/8969482/122180719-97a15300-ce80-11eb-9bc9-4d20ce10daf2.png)

It seems to have run ok up until this point. I have indel_region.MAF files and e.g. the indel_features.pdf looks sensible. Any thoughts what might be happening or how I can figure it out?

Thanks a lot

Dan

**Comment by AmandaMilou** [comment]
Hi Dan,
It seems like some of the predicted mutation probabilities are 0 or 1, making the final p-value/ log p-value undefined. You may try to re-run the model with fewer features. We will work on updating the code to go catch the error when this happens.

**Comment by danchubb** [comment]
Hi Amanda,

Thanks for the quick response. I think I follow that. I've tried removing all features other than the ones the method calculates automatically:

![image](https://user-images.githubusercontent.com/8969482/122230208-537a7680-ceb1-11eb-8875-e4cee5099c98.png)


 however I am still getting:

![image](https://user-images.githubusercontent.com/8969482/122229967-1f06ba80-ceb1-11eb-9234-e498f23cdf06.png)

Thanks

Dan

**Comment by AmandaMilou** [comment]
HI Dan,

I think you can try to reduce the number of features further by adjusting the threshold in feature selection. See section https://github.com/skandlab/MutSpot#adjusting-threshold

**Comment by danchubb** [comment]
Thanks, Amanda

I have ran from scratch with no additional epigenetic or sample features and the parameters : cutoff.nucleotide.new=0.93 and top.nucleotide=3

Am I right that that should only choose the top 3 features? So in the example I pasted above that would be Polya, local mutation rate and polyg?

And I get the same error. Should I be even more conservative? 

I'm not 100% on what's happening here. why would more features result in a mutation probability of 0 and 1? 

Thanks again,

Dan

**Comment by AmandaMilou** [comment]
Hi Dan,

You can try to trim the polyG feature and fit the model again. Mutation probability of 0 or 1 means the model can separate the positive and negative cases completely given a feature combination. i.e. for a particular combination of features, all the sites in your data are mutated or not mutated. You can try to avoid it by reducing the number of features. This may also happen when there is too few mutations in your data and there is not enough variability in the data.

**Comment by danchubb** [comment]
HI Amanda,

Apologies for the delay. Sorry, when you say trim the polyG feature, do you mean removing it from the model? Can you suggest what command I'd use to do that? Would for example just using top.nucleotide=2 just do the trick? I'll try it now anyway and see what I get. 

Thanks

Dan

**Comment by danchubb** [comment]
Hi, I have reran with top.nuclotide=2 and am still getting the error with indels. 

I don't know if it's a problem with low numbers of mutations : 

looking at the indel_region.maf I have:

1446 samples
227688 mutations from 176925 sites, does that seem reasonable? 

Thanks

Dan

**Comment by danchubb** [comment]
HI, I've tried several different bed regions and one that contains the core promoter regions doesn't crash. 

The indel_region.maf is much smaller though, 

2132 mutations across 1863 unique sites in 1062 samples. 

Could the size be a problem?

Is there anything I can do to compare the runs? 

Thanks

Dan

**Comment by danchubb** [comment]
Hi,

I confirmed that in at least one case it was being caused by a pvalue of 0 in the results.  This particular mutation was in a large proportion of cases and was likely to be a false positive that remained after filtering. 

I have added a line in the mutPredict.indel.run.lr function between

184  results = results[ ,-1, drop = FALSE]
185  class(results) <- "numeric"

results[results[,1]==0, 1]=1E-99

(I tried .Machine$double.xmin but it really broke the manhattan plot y axis!)

So it remains highly significant and doesn't break the results anymore. Does that seem like a reasonable fix?

Cheers,

Dan
